src/sdk/pynni/nni/hyperopt_tuner/hyperopt_tuner.py

```diff
@@ -286,3 +277,26 @@ def miscs_update_idxs_vals(self, miscs, idxs, vals,
                 if assert_all_vals_used or tid in misc_by_id:
                     misc_by_id[tid]['idxs'][key] = [tid]
                     misc_by_id[tid]['vals'][key] = [val]
+    
+    def update_parameters_queue(self):
+        '''update the parameters queue when it is empty or when one trial result is received
+        '''
+        rval = self.rval
+        trials = rval.trials
+        algorithm = rval.algo
+        new_ids = rval.trials.new_trial_ids(self.max_queue_len)
+        rval.trials.refresh()
+        random_state = rval.rstate.randint(2**31-1)
+        new_trials = algorithm(new_ids, rval.domain, trials, random_state)
+        rval.trials.refresh()
+        for item in new_trials:
+            vals = item['misc']['vals']
+            parameter = dict()
+            for key in vals:
+                try:
+                    parameter[key] = vals[key][0].item()
+                except Exception:
+                    parameter[key] = None
```

Comment:
log exception here and only catch expected Exception type.

Reply:
This is the origin code and I think it reasonable.